### PR TITLE
feat(repo): add repo counter sync with build number

### PR DESCRIPTION
**Pull request by kneal**
PR includes the following:

* Updates to readme/docs/make for altering tables to new counter field
* Updates to readme/docs/make for syncing counter field with build number
* Added migration code for altering table for counter field
* Added migration code for syncing the build number field to repo counter

Related to: https://github.com/go-vela/community/issues/137

Depends on: https://github.com/go-vela/types/pull/176 _should fail without this change in the mod_

### CODE REVIEW

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by meinenec — APPROVED** [review]

File: migrations/v0.8/DOCS.md [hunk 1]
```diff
@@ -139,6 +139,110 @@ make run
 
 > NOTE: Please review the [start section](#start) before moving forward.
 
-This utility supports invoking the following actions when migrating to `v0.8.x`:
+This utility supports invoking the following actions when migrating to `v0.7.x`:
```
Comment (wass3r) [inline migrations/v0.8/DOCS.md:142]: shouldn't this stay at `v0.8.x`?

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**